cfgov/jinja2/v1/_includes/leadership-calendar-table.html

```diff
@@ -1,33 +1,59 @@
-{% macro render(user_options={}) %}
-
 {% import 'macros/time.html' as time %}
 
+{% macro render(user_options={}, is_pdf=false) %}
+
 {% set options = {
     'time_col_classes': 'u-w25pct',
     'name_col_classes': 'u-w25pct',
     'desc_col_classes': 'u-w50pct'
 } %}
 {% set _ignore = options.update(user_options) %}
+{% set paged_events = none %}
+
+{% if not is_pdf %}
+    {# Page the events by the earliest and latest dates in the range. #}
+    {% set range = day_range or page_days %}
+    {% set from = range[-1] %}
+    {% set to = range[0] %}
+    {% set paged_events = events.filter(dtstart__gte=from).filter(dtstart__lte=to).order_by('-dtstart') %}
+{% endif %}
+
+{% if is_pdf %}
+    {% for days_of_events in events|groupby('day')|reverse %}
+        {{ render_table(days_of_events, is_pdf, options) }}
+    {% endfor %}
+{% else %}
+    {% for days in page_days %}
+        {{ render_table(days, is_pdf, options) }}
+    {% endfor %}
+{% endif %}
 
-{# Page the events by the earliest and latest dates in the range. #}
-{% set range = day_range or page_days %}
-{% set from = range[-1] %}
-{% set to = range[0] %}
-{% set paged_events = events.filter(dtstart__gte=from).filter(dtstart__lte=to).order_by('-dtstart') %}
+{% endmacro %}
+
+{% macro render_table(days, is_pdf, options) %}
+    <table class="u-w100pct block block__flush-top">
+        <thead>
+            <tr>
+                <th colspan="3">
+                    {% set day = days.grouper if is_pdf else days %}
+                    {# {% import 'macros/time.html' as time %} #}
+                    {{ time.render(day, {'date':true}) }}
+                </th>
+            </tr>
+        </thead>
+        <tbody>
+            {{ render_events(events, day, options) }}
+        </tbody>
+    </table>
+{% endmacro %}
 
-{% for day in page_days %}
-<table class="u-w100pct block block__flush-top">
-    <thead>
-        <tr>
-            <th colspan="3">
-                {{ time.render(day, {'date':true}) }}
-            </th>
-        </tr>
-    </thead>
-    <tbody>
-    {% for event in paged_events|sort(attribute='dtstart') %}
+{% macro render_events(calendar_events, day, options) %}
+    {% if day.date is defined %}
+        {% set day = day.date() %}
+    {% endif %}
+    {% for event in calendar_events|sort(attribute='dtstart') %}
```

Comment:
Can you sort before you get here? This is considered poor form to sort in the template.